src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt

```diff
@@ -20,20 +21,22 @@ class NatsParkingUpdateController(
     override val parser: Parser<ParkingOuterClass.UpdateParkingRequest> =
         ParkingOuterClass.UpdateParkingRequest.parser()
 
-    override fun generateReplyForNatsRequest(
-        request: ParkingOuterClass.UpdateParkingRequest
-    ): ParkingOuterClass.UpdateParkingResponse {
+    override fun generateReplyForNatsRequest(request: ParkingOuterClass.UpdateParkingRequest): Mono<ParkingOuterClass.UpdateParkingResponse> {
         val parking =
             converter
                 .protoToParking(request.parking)
                 .apply { id = request.parkingId }
-        val createdParking = service.updateEntity(parking)
-
-        return ParkingOuterClass.UpdateParkingResponse.newBuilder()
-            .setParking(
-                converter
-                    .parkingToProto(createdParking)
-            )
-            .build()
+
+        return Mono.fromCallable {
+            val createdParking = service.updateEntity(parking)
```

Comment:
it's better not to execute some logic outside the reactor chain, then when something fails, an error will be propagated in the reactor chain. so move your protoToParking mapping to reactor chain

Reply:
✅